clang/lib/CodeGen/ItaniumCXXABI.cpp

```diff
@@ -3565,8 +3565,30 @@ void ItaniumRTTIBuilder::BuildVTablePointer(const Type *Ty) {
   // Check if the alias exists. If it doesn't, then get or create the global.
   if (CGM.getItaniumVTableContext().isRelativeLayout())
     VTable = CGM.getModule().getNamedAlias(VTableName);
-  if (!VTable)
-    VTable = CGM.getModule().getOrInsertGlobal(VTableName, CGM.Int8PtrTy);
+
+  // To generate valid device code SPIR-V module with virtual functions these
+  // global pointers should have global address space, not private as by default
+  auto VTableTy =
+      CGM.getLangOpts().SYCLIsDevice &&
+              CGM.getLangOpts().SYCLAllowVirtualFunctions &&
+              (VTableName == ClassTypeInfo || VTableName == SIClassTypeInfo)
+          ? CGM.Int8Ty->getPointerTo(
+                static_cast<unsigned>(LangAS::opencl_global))
+          : CGM.Int8PtrTy;
+  if (!VTable) {
+    if (VTableTy == CGM.Int8PtrTy) {
+      VTable = CGM.getModule().getOrInsertGlobal(VTableName, VTableTy);
+    } else {
+      VTable = CGM.getModule().getOrInsertGlobal(VTableName, VTableTy, [&] {
+        return new llvm::GlobalVariable(
+            CGM.getModule(), VTableTy, false,
```

Comment:
Please add a comment explaining what `false` parameter means. 
```suggestion
            CGM.getModule(), VTableTy, false,
```
https://llvm.org/docs/CodingStandards.html#comment-formatting for info.

Reply:
Done & also for the other [c810c23](https://github.com/intel/llvm/pull/7255/commits/c810c2359c2ab52a9992d5d1c206155a3f0514bc)